src/number_theory/kummer_dedekind.lean

```diff
@@ -52,20 +52,157 @@ with a formula).
+    { rw mul_comm,
+      exact mem_conductor_iff.mp (ideal.mem_comap.mp hp) _ },
+    refine ⟨_, by simpa only [ring_hom.map_mul, mul_comm (algebra_map R S p) (l a)]⟩,
+    rw mul_comm,
+    apply ideal.mul_mem_left (I.map (algebra_map R (R[x]))) _
+      (ideal.mem_map_of_mem _ ha) },
+  refine finset.sum_induction _ (λ u, u ∈ (algebra_map (R[x]) S) ''
+    (I.map (algebra_map R (R[x]))))
+    (λ a b, _) _ _,
+  rintro ⟨z, hz, rfl⟩ ⟨y, hy, rfl⟩,
+  rw [← ring_hom.map_add],
+  exact ⟨z + y, ideal.add_mem _ (set_like.mem_coe.mp hz) hy, rfl⟩,
+  { refine ⟨0, set_like.mem_coe.mpr $ ideal.zero_mem _, ring_hom.map_zero _⟩ },
+  { intros y hy,
+    exact lem ((finsupp.mem_supported _ l).mp H hy) },
+end
+
+/-- A technical result telling us that `(I * S) ∩ R[x] = I * R[x]` for any ideal `I` of `R`. -/
+lemma comap_map_eq_map_adjoin_of_coprime_conductor
+  (hx : (conductor R x).comap (algebra_map R S) ⊔ I = ⊤)
+  (h_alg : function.injective (algebra_map (R[x]) S)):
+  (I.map (algebra_map R S)).comap (algebra_map (R[x]) S) = I.map (algebra_map R (R[x])) :=
+begin
+  apply le_antisymm,
+  { -- This is adapted from [Neukirch1992]. Let `C = (conductor R x)`. The idea of the proof
+    -- is that since `I` and `C ∩ R` are coprime, we have
+    -- `(I * S) ∩ R[x] ⊆ (I + C) * ((I * S) ∩ R[x]) ⊆ I * R[x] + I * C * S ⊆ I * R[x]`.
+    intros y hy,
+    obtain ⟨z, hz⟩ := y,
+    obtain ⟨p, hp, q, hq, hpq⟩ := submodule.mem_sup.mp ((ideal.eq_top_iff_one _).mp hx),
+    have temp : (algebra_map R S p)*z + (algebra_map R S q)*z = z,
+    { simp only [←add_mul, ←ring_hom.map_add (algebra_map R S), hpq, map_one, one_mul] },
+    suffices : z ∈ algebra_map (R[x]) S '' (I.map (algebra_map R (R[x]))) ↔ (⟨z, hz⟩ : R[x]) ∈
+      I.map (algebra_map R (R[x])),
+    { rw [← this, ← temp],
+      obtain ⟨a, ha⟩ := (set.mem_image _ _ _).mp (prod_mem_ideal_map_of_mem_conductor hp
+        (show z ∈ I.map (algebra_map R S), by rwa ideal.mem_comap at hy )),
+      use a + (algebra_map R (R[x]) q) * ⟨z, hz⟩,
+      refine ⟨ ideal.add_mem (I.map (algebra_map R (R[x]))) ha.left _,
+        by simpa only [ha.right, map_add, alg_hom.map_mul, add_right_inj] ⟩,
+      rw mul_comm,
+        exact ideal.mul_mem_left (I.map (algebra_map R (R[x]))) _ (ideal.mem_map_of_mem _ hq) },
+    refine ⟨ λ h, _, λ h, (set.mem_image _ _ _).mpr (exists.intro ⟨z, hz⟩ ⟨by simp [h], rfl⟩ ) ⟩,
+    { obtain ⟨x₁, hx₁, hx₂⟩ := (set.mem_image _ _ _).mp h,
+      have : x₁ = ⟨z, hz⟩,
+      { apply h_alg,
+        simpa [hx₂], },
+      rwa ← this }  },
+
+  { -- The converse inclusion is trivial
+    have : algebra_map R S = (algebra_map _ S).comp (algebra_map R (R[x])) := by { ext, refl },
+    rw [this, ← ideal.map_map],
+    apply ideal.le_comap_map }
+end
+
+/-- The canonical morphism of rings from `R[x] ⧸ (I*R[x])` to `S ⧸ (I*S)` is an isomorphism
+    when `I` and `(conductor R x) ∩ R` are coprime. -/
+noncomputable def quot_adjoin_equiv_quot_map (hx : (conductor R x).comap (algebra_map R S) ⊔ I = ⊤)
```

Comment:
Maybe a `simps` tag? It depends on what you want to do with this map, but we probably want at least a minimal API around it.

Reply:
The I tried adding `simps apply` but it gave the "wrong" projection lemma so it added a projection lemma manually.

Reply:
In a later PR - possible #16870  - I'll prove the usual criterion for the Dedekind-Kummer theorem in terms of the divisibility of the index `[O_K : Z[a] ]` by the prime ideal `I` so I'll probably end up adding more API to do that.